lib/App/Prove/State.pm

```diff
@@ -390,7 +390,24 @@ sub _get_raw_tests {
         if ( '-' eq $arg ) {
             local $/;
             my $in = <STDIN>;
-            if ($in =~ /^(?:1\.\.\d|ok|not ok)\b(?!\S)/m) {
+            if ($in =~ /
+                \A                   # Beginning of first line
+                TAP\s+version\s+\d+  # TAP version line
+                |
+                ^                    # Beginning of any line
+                \s*                  # Indented subtests
+                (?:
+                    [#]              # A comment
+                    |
+                    (?:
+                        1\.\.\d |    # A plan line
```

Comment:
I think that should be `\d+`, otherwise it would fail to match `1..10`

Reply:
Code is only as good as tests.  Good catch.